bangazonapi/views/paymenttype.py

```diff
@@ -80,11 +80,14 @@ def destroy(self, request, pk=None):
     def list(self, request):
         """Handle GET requests to payment type resource"""
         payment_types = Payment.objects.all()
-
-        customer_id = self.request.query_params.get('customer', None)
+        
+        try:
+            customer_id = Customer.objects.get(user=request.auth.user)
+        except AttributeError:
+            customer_id = None
 
         if customer_id is not None:
-            payment_types = payment_types.filter(customer__id=customer_id)
+            payment_types = payment_types.filter(customer__id=customer_id.id)
```

Comment:
I am asking since `customer_id.id` seems redundant.

Reply:
I was just using the variable that was already in the code as I'd have to change it other places as well. I can do that.